datalake/src/main/java/com/sequenceiq/datalake/flow/dr/validation/handler/DatalakeBackupValidationWaitHandler.java

```diff
@@ -0,0 +1,83 @@
+import com.sequenceiq.datalake.service.sdx.PollingConfig;
+import com.sequenceiq.datalake.service.sdx.dr.SdxBackupRestoreService;
+import com.sequenceiq.flow.event.EventSelectorUtil;
+import com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler;
+import com.sequenceiq.flow.reactor.api.handler.HandlerEvent;
+
+import reactor.bus.Event;
+
+@Component
+public class DatalakeBackupValidationWaitHandler extends ExceptionCatcherEventHandler<DatalakeBackupValidationWaitRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatalakeBackupValidationWaitHandler.class);
+
+    @Value("${sdx.stack.backup.status.sleeptime_sec:20}")
+    private int sleepTimeInSec;
+
+    @Value("${sdx.stack.backup.status.duration_min:240}")
+    private int durationInMinutes;
+
+    @Inject
+    private SdxBackupRestoreService sdxBackupRestoreService;
+
+    @Override
+    public String selector() {
+        return EventSelectorUtil.selector(DatalakeBackupValidationWaitRequest.class);
+    }
+
+    @Override
+    protected Selectable defaultFailureEvent(Long resourceId, Exception e, Event<DatalakeBackupValidationWaitRequest> event) {
+        return new DatalakeBackupValidationFailedEvent(resourceId, null, e);
+    }
+
+    @Override
+    protected Selectable doAccept(HandlerEvent<DatalakeBackupValidationWaitRequest> event) {
+        DatalakeBackupValidationWaitRequest request = event.getData();
+        Long sdxId = request.getResourceId();
+        String userId = request.getUserId();
+        Selectable response;
+        try {
+            LOGGER.info("Start polling datalake full backup status for id: {}", sdxId);
+            PollingConfig pollingConfig = new PollingConfig(sleepTimeInSec, TimeUnit.SECONDS, durationInMinutes,
+                TimeUnit.MINUTES);
+            sdxBackupRestoreService.waitForDatalakeDrBackupToComplete(sdxId, request.getOperationId(), request.getUserId(),
+                pollingConfig, "Backup validation");
+            response = new SdxEvent(DATALAKE_BACKUP_VALIDATION_SUCCESS_EVENT.event(), sdxId, userId);
+        } catch (UserBreakException userBreakException) {
+            LOGGER.info("Backup validation polling exited before timeout. Cause: ", userBreakException);
+            response = new DatalakeBackupValidationFailedEvent(sdxId, userId, userBreakException);
+        } catch (PollerStoppedException pollerStoppedException) {
+            LOGGER.info("Backup validation poller timed out for cluster: {}", sdxId);
+            response = new DatalakeBackupValidationFailedEvent(sdxId, userId,
+                new PollerStoppedException("Datalake backup validation timed out after " + durationInMinutes + " minutes"));
+        } catch (PollerException exception) {
+            LOGGER.info("Backup validation polling failed for cluster: {}", sdxId);
+            response = new DatalakeBackupValidationFailedEvent(sdxId, userId, exception);
```

Comment:
This will be the path that all validation errors will follow, Do we want to extract a more user friendly error message?

Reply:
This is converted later into a more user-friendly message using the exception type.